### PR TITLE
[MM-20353] Change KVSetWithOptions to accept a byte slice

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Because we can't send an arbitrary interface via RPC we need change `KVSetWithOptions` to accept a byte slice instead of an interface. This also means we can drop `EncodeJSON` and do JSON encoding on the client side.

The PR also includes the unit tests that @ilgooz has written for https://github.com/mattermost/mattermost-server/pull/13132. Even though the code change in his PR is not longer needed, the unit tests are still valuable.

cc @ilgooz

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-20353
- https://mattermost.atlassian.net/browse/MM-16822

### CODE REVIEW

File: app/plugin_test.go [hunk 1]
```diff
@@ -197,135 +198,118 @@ func TestPluginKeyValueStoreSetWithOptionsJSON(t *testing.T) {
...
-		assert.Equal(t, []byte(nil), ret)
-	})
-
-	t.Run("storing a value json encoded works", func(t *testing.T) {
-		result, err := th.App.SetPluginKeyWithOptions(pluginId, "key", map[string]interface{}{
-			"val-a": 10,
-		}, model.PluginKVSetOptions{
-			EncodeJSON: true,
-		})
+	t.Run("storing a value works without providing options works", func(t *testing.T) {
```
Comment (lieut-data) [inline app/plugin_test.go]:
```suggestion
	t.Run("storing a value without providing options works", func(t *testing.T) {
```

**Comment by prapti** [comment]
@hanzei there are two Jira tickets linked to this PR. Would it be possible to add some test steps so I can run through them quick? That would be great.

**Comment by hanzei** [comment]
@prapti There was no process made on setting up an example usage of the KV store for the demo plugin. Given that the demo plugin doesn't use the KV at all, it will take some mana to design and implement it. I can fill a (maybe HW) ticket next week, but that won't help for this PR.


Similar to https://github.com/mattermost/mattermost-server/pull/13041 the best way to test this, is by writing some code for a plugin and use the method.

Let me know, if this helps.

**Pull request merged**